### PR TITLE
build: releases using goreleaser

### LINKED ISSUE #24: Automated releases
**What would you like to be added**:
Instead of doing releases manually, we want do to them automatically.
Goreleaser does that job very well.

Goreleaser website:
https://goreleaser.com/

See kubectl-trace as an example to see how it's configured:
https://github.com/iovisor/kubectl-trace/blob/master/.goreleaser.yml

**Why is this needed**:

Manual releases are a pain

### PR DESCRIPTION
**Pull request by fntlnz**
Signed-off-by: Lorenzo Fontana <lo@linux.com>

**What type of PR is this?**

/kind documentation
/kind feature

**Any specific area of the project related to this PR?**
NONE

**What this PR does / why we need it**:

We want to have all the releases done in the same way.
Using goreleaser we can produce the same artifacts regardless who is the maintainer doing the release by minimizing the manual steps. This also adds documentation for the single manual step we have to do

In the future this manual step can be automated by a CI when this project will have one.

**Which issue(s) this PR fixes**:

Fixes #24 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: release.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+```bash
+git tag -a v0.1.0-rc.0 -m "v0.1.0-rc.0"
+git push origin v0.1.0-rc.0
+```
+
+Run goreleaser, make sure to export your GitHub token first.
+
+```
+export GITHUB_TOKEN=<YOUR_GH_TOKEN>
+make release
```
Comment (Issif) [inline release.md]: where's Makefile?

Comment (fntlnz) [reply release.md]: I didn't do the makefile in the end! Updating this! thanks

File: release.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# falcoctl Release Process
```
Comment (leodido) [inline release.md]:
```suggestion
# Release Process
```

Redundant.

File: release.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Our release process is automated using [goreleaser](https://github.com/goreleaser/goreleaser).
+
+When we release we do the following process:
+
+1. We decide together (usually in the #falco channel in Sysdig's slack) what's the next version to tag
```
Comment (leodido) [inline release.md]:
```suggestion
1. We decide together (usually in the #falco channel in [slack](https://sysdig.slack.com)) what's the next version to tag
```

**Pull request merged**